remotespark/livyclientlib/livysession.py

```diff
@@ -106,11 +107,38 @@ def create_sql_context(self):
         else:
             raise FailedToCreateSqlContextException("Failed to create the SqlContext.\nError: '{}'".format(out))
 
+    def get_app_id(self):
+        if self._app_id is None:
+            session = self._http_client.get_session(self.id)
+            self._app_id = session['appId']
```

Comment:
There's no Livy versioning at this moment... What we should do is gracefully handle absences at the very least.